src/vos/vos_dtx.c

```diff
@@ -31,8 +31,41 @@
 #include "vos_layout.h"
 #include "vos_internal.h"
 
-#define DTX_UMMID_ABORTED	((umem_id_t){ .pool_uuid_lo = -1, .off = -1, })
-#define DTX_UMMID_UNKNOWN	((umem_id_t){ .pool_uuid_lo = -1, .off = -2, })
+static inline bool
+dtx_is_null(umem_off_t umoff)
+{
+	/* If @umoff is zero, then it will be regared as NULL case.
+	 * That is compatible with the case of NOT set DTX things.
+	 *
+	 * In theory, zero is valid offset in a PMEM pool.
+	 * UMMID_IS_NULL() also handles zero offset as NULL.
+	 */
+	return umoff == UMOFF_NULL || umoff == 0;
```

Comment:
In the case of UMMID, you also have the pool flag to indicate NULL.   This is probably OK in practice but in theory a 0 offset could be a valid offset.   Though if PMDK always uses 0 for internal use, it probably is OK to treat it as NULL.   One alternative would be to change UMOFF_IS_NULL would compare <= FLAG_MAX (or whatever it's called) instead of using high end bits and just assume that a real 0 offset can't happen (since we always do allocation using umem_alloc, we could assert if this ever happens).  UMOFF_NULL could be set to 0 in this case.

I still think it's safest to not assume it's 0 and to initialize it and I don't think it adds significant cost to do so since we already have to log the node we are modifying, no?

Reply:
I will check whether only "umoff == UMOFF_NULL" works or not.